checks.d/spark.py

```diff
@@ -71,27 +71,42 @@
 '''
 
 # stdlib
+import time
 from urlparse import urljoin, urlsplit, urlunsplit
 
 # 3rd party
 import requests
 from requests.exceptions import Timeout, HTTPError, InvalidURL, ConnectionError
 from simplejson import JSONDecodeError
+from bs4 import BeautifulSoup
 
 # Project
 from checks import AgentCheck
 
+# Switch that determines the mode Spark is running in. Can be either
+# 'yarn' or 'standalone'
+SPARK_CLUSTER_MODE = 'spark_cluster_mode'
+SPARK_STANDALONE_MODE = 'spark_standalone_mode'
+SPARK_YARN_MODE = 'spark_yarn_mode'
+
+SPARK_STANDALONE_MASTER = 'spark_standalone_master_uri'
+SPARK_STANDALONE_SERVICE_CHECK = 'spark_standalone_master'
```

Comment:
this should probably be something along the lines of `spark.standalone_master.can_connect`